packages/start-plugin-core/src/start-router-plugin/plugin.ts

```diff
@@ -14,20 +14,33 @@ import {
   tanstackRouterAutoImport,
   tanstackRouterGenerator,
 } from '@tanstack/router-plugin/vite'
+import { VITE_ENVIRONMENT_NAMES } from '../constants'
 import { routeTreeClientPlugin } from './route-tree-client-plugin'
 import { virtualRouteTreePlugin } from './virtual-route-tree-plugin'
 import { routesManifestPlugin } from './generator-plugins/routes-manifest-plugin'
 import { serverRoutesPlugin } from './generator-plugins/server-routes-plugin'
 import type { PluginOption } from 'vite'
-import type { Config } from '@tanstack/router-generator'
+import type { Config } from '@tanstack/router-plugin'
 
 export function tanStackStartRouter(config: Config): Array<PluginOption> {
   return [
     tanstackRouterGenerator({
       ...config,
       plugins: [serverRoutesPlugin(), routesManifestPlugin()],
+      environmentName: VITE_ENVIRONMENT_NAMES.client,
```

Comment:
Rename this to to something like `viteEnvironmentName` or something like `viteApplyEnvironment`?

Purely since it wouldn't have any effect on the other bundlers.